src/zarr/core/array.py

```diff
@@ -3448,4 +3492,333 @@ def _get_default_codecs(
     else:
         dtype_key = "numeric"
 
-    return [{"name": codec_id, "configuration": {}} for codec_id in default_codecs[dtype_key]]
+    return default_codecs[dtype_key]
+
+
+FiltersParam: TypeAlias = (
+    Iterable[dict[str, JSON] | Codec] | Iterable[numcodecs.abc.Codec] | Literal["auto"]
+)
+CompressionParam: TypeAlias = (
+    Iterable[dict[str, JSON] | Codec] | Codec | numcodecs.abc.Codec | Literal["auto"]
+)
+
+
+async def create_array(
+    store: str | StoreLike,
+    *,
+    name: str | None = None,
+    shape: ShapeLike,
+    dtype: npt.DTypeLike,
+    chunks: ChunkCoords | Literal["auto"] = "auto",
+    shards: ChunkCoords | Literal["auto"] | None = None,
+    filters: FiltersParam = "auto",
+    compressors: CompressionParam = "auto",
+    fill_value: Any | None = 0,
+    order: MemoryOrder | None = "C",
+    zarr_format: ZarrFormat | None = 3,
+    attributes: dict[str, JSON] | None = None,
+    chunk_key_encoding: ChunkKeyEncoding | ChunkKeyEncodingParams | None = None,
+    dimension_names: Iterable[str] | None = None,
+    storage_options: dict[str, Any] | None = None,
+    overwrite: bool = False,
+    config: ArrayConfig | ArrayConfigParams | None = None,
+    data: npt.ArrayLike | None = None,
```

Comment:
how important is it that we include the `data` kwarg. Previously, if `data` was provided, shape/dtype were inferred from the input. That's now how things work here (which is fine) but it draws into question the value of this parameter.

I suggest we remove this in favor of a `zarr.from_array` function (to come later).

Reply:
+1, i think data doesn't fit here, and `from_array` is a better spot for it

Reply:
`data` is removed as a kwarg in https://github.com/zarr-developers/zarr-python/pull/2463/commits/77f40a5de0422ff2f14ae55d54e8b94637f870a5